.github/workflows/osx.yml

```diff
@@ -0,0 +1,32 @@
+name: OSX-build
+
+on: [push]
+
+jobs:
+  osx-clang-omp:
+    strategy:
+      fail-fast: false
+      matrix:
+        config:
+        - {shared: "ON", build_type: "Debug", name: "omp/debug/shared"}
+        - {shared: "OFF", build_type: "Release", name: "omp/release/static"}
+    name: ${{ matrix.config.name }}
+    runs-on: [macos-latest]
+
+    steps:
+    - uses: actions/checkout@v2
+    - name: setup
+      run: brew install libomp
+    - name: info
+      run: |
+        g++ -v
+        cmake --version
+    - name: configure
+      run: |
+        mkdir build
+        cd build
+        cmake .. -DBUILD_SHARED_LIBS=${{ matrix.config.shared }}
+        cmake --build . -j8 --config ${{ matrix.config.build_type }}
+        ctest . -C ${{ matrix.config.build_type }}
```

Comment:
cmake --build . -j8 --config build_type should work for mult-configuration toolset like MSVC or XCode. If you use gcc/clang, using the previous compile step sets correct build type.
I think clang from XCode is the type of clang not XCode
from `2020-03-04T12:25:58.9740980Z -- Setting build type to 'Release' as none was specified.` in [omp/debug/shared log](https://pipelines.actions.githubusercontent.com/dR6M3a2IKUyMCOpbBJ1SfpHlfxqOoIda0x3iCmpxIvMt04aaNr/_apis/pipelines/1/runs/35/signedlogcontent/3?urlExpires=2020-03-04T13%3A00%3A14.8711618Z&urlSigningMethod=HMACV1&urlSignature=DES9JRSAK8vmpn2EajqyLA18SWAhl1Ge9B6DnbkszxM%3D)